docs/design/expressions/member_access.md

```diff
@@ -841,6 +842,56 @@ alias X3 = (i32 as Factory).Make;
 alias X4 = i32.((i32 as Factory).Make);
 ```
 
+## Tuple indexing
+
+A tuple indexing expression is of the form:
+
+-   _expression_ `.` _integer-literal_
+-   _expression_ `->` _integer-literal_
+
+The _expression_ is required to be of tuple type.
```

Comment:
Presumably this only applies to the `.` form, and the _expression_ in the `->` form is required to be of type pointer to tuple?

There's a similar problem on line 878.